core/arch/x86/encode.c

```diff
@@ -1529,6 +1529,9 @@ encoding_possible_pass(decode_info_t *di, instr_t *in, const instr_info_t *ii)
     opnd_t using_vvvv_bits = opnd_create_null();
     opnd_t using_aaa_bits = opnd_create_null();
 
+    di->requires_vsib_ymm = TEST(REQUIRES_VSIB_YMM, ii->flags);
+    di->requires_vsib_zmm = TEST(REQUIRES_VSIB_ZMM, ii->flags);
```

Comment:
Hmm, now I see what you were saying: somehow I thought this would be done just once per instr in instr_encode_arch, not once per considered template.  Maybe how you had it originally with passing the flags is better.

Reply:
(Reverted)